### PR TITLE
Pull variables out of anti-virus script so that script can be reused in other repos

### PR DESCRIPTION
**Pull request by chrisgilmerproj**
## Description

The anti-virus script needs to be available to other repos which means that configuration must be done with environment variables instead of hard-coding values. This PR updates the script to remove all transcom/mymove specific references. A future update will probably move this entire script to a Truss owned repo.

Primarily you're going to see environment variable changes here. I've also added docs that print out during failures.

This work supports https://github.com/transcom/milmove_orders/pull/53

## Setup

```sh
direnv allow
make clean
make anti_virus
```

### CODE REVIEW

**Comment by chrisgilmerproj** [comment]
Proof that this works in CircleCI: https://circleci.com/gh/transcom/mymove/248526

**Comment by chrisgilmerproj** [comment]
Proof the last changes worked: https://circleci.com/gh/transcom/mymove/248709

File: .circleci/config.yml [hunk 1]
```diff
@@ -388,7 +388,18 @@ jobs:
       - checkout
       - setup_remote_docker:
           docker_layer_caching: false
-      - run: make anti_virus
+      - run:
+          name: make anti_virus
+          command: make anti_virus
+          environment:
+            AV_DIR: /home/circleci/transcom/mymove
```
Comment (mr337) [inline .circleci/config.yml]: Are these the minimum ENVARS needed to run? If so would it be handy to have an example of the EVNAR values at the top of the `anti-virus` script?

File: scripts/anti-virus [hunk 2]
```diff
@@ -15,28 +28,41 @@ function msg_now() {
...
+MOUNT_DIR=${AV_MOUNT_DIR:-/project}
 readonly MOUNT_DIR
-IGNORE_FILES=(pkg/testdatagen/testdata/orders.pdf)
+IGNORE_FILES=${AV_IGNORE_FILES:-}
 readonly IGNORE_FILES
-# WARNING: IGNORE AT OUR PERIL. IF ADDING HERE ADD NOTES!
-# - PUA.Pdf.Trojan.EmbeddedJavaScript-1 is ignored because we don't ship PDFs in any docker containers
-# - orders.pdf.UNOFFICIAL is a finding based on the ignored file above of the same name
-IGNORE_SIGS=(PUA.Pdf.Trojan.EmbeddedJavaScript-1 orders.pdf.UNOFFICIAL)
+IGNORE_SIGS=${AV_IGNORE_SIGS:-}
```
Comment (mr337) [inline scripts/anti-virus]: This is me being nitpicky and won't block this. 

Would printing out an thing saying these are non empty be beneficial?

File: scripts/anti-virus [hunk 3]
```diff
@@ -121,17 +165,20 @@ fi
 # See https://www.clamav.net/documents/file-hash-signatures
 echo
 echo "Creating whitelist of files to ignore:"
+
+rm -f "${WHITELIST_FILES}"
```
Comment (mr337) [inline scripts/anti-virus]: I forgot all about the `-f` trick when using rm.

```
$ rm /tmp/blah.log
rm: cannot remove '/tmp/blah.log': No such file or directory
$ echo $?
1

$ rm -f /tmp/blah.log
$ echo $?
0
```

File: scripts/anti-virus [hunk 4]
```diff
@@ -64,25 +90,41 @@ docker pull "${DOCKER_IMAGE}"
...
+    echo " - For FreshClam Errors: https://www.clamav.net/documents/freshclam-faq"
+    echo " - For ClamAV Troubleshooting: https://www.clamav.net/documents/troubleshooting-faq"
+    echo " - For other questions: https://www.clamav.net/documents/clam-antivirus-user-manual"
+    echo "********************************************************************************"
+    if [ "${COUNT}" == "5" ]; then
       echo "Script cannot continue now that retries are exhuasted"
       exit 1
     else
-      break
+      continue
```
Comment (mr337) [inline scripts/anti-virus]: This else block _feels_ redundant.

File: scripts/anti-virus (hunk 2 above)
Comment (chrisgilmerproj) [reply scripts/anti-virus]: I do print them out later on. Is that what you're looking for?

File: scripts/anti-virus (hunk 4 above)
Comment (chrisgilmerproj) [reply scripts/anti-virus]: it kinda is, but for clarity I had it.

File: .circleci/config.yml (hunk 1 above)
Comment (chrisgilmerproj) [reply .circleci/config.yml]: They aren't the minimum. The only thing you have to provide is `AV_DIR` and the script will tell you if its wrong.

Comment (mr337) [reply .circleci/config.yml]: What I get commenting before reading the whole PR.

File: scripts/anti-virus (hunk 2 above)
Comment (mr337) [reply scripts/anti-virus]: More as a guard to make sure someone doesn't inadvertently  set it and forget which shouldn't happen in circleci. Disregard!

**Pull request merged**